server/store/oauth2_store.go

```diff
@@ -18,11 +23,11 @@ func (s *pluginStore) VerifyOAuth2State(state string) error {
 		return err
```

Comment:
@avddvd Moving the conversation on the issue https://github.com/mattermost/mattermost-plugin-mscalendar/issues/71#issuecomment-615768566 to here since the code is here.

I could be entirely wrong but I think `string(data) != state` would occur if someone is impersonating Microsoft or the user tries to authenticate twice or something. That error can stay the way that you currently have it :+1:

If `err == store.ErrNotFound` in the block below, then we know specifically that the state is not in store, and can assume this means the Expiry has occurred. So, my only suggestion is that we add this block:

```suggestion
		if err == store.ErrNotFound {
			return errors.new("authentication attempt expired, please try again")
		}
		return err
```

Reply:
Thanks @mickmister for the clarification, pushed a new change to check error in case of key not found (i.e: since TTL expired )

Reply:
This `err == store.ErrNotFound` check needs to be in the `err != nil` block as my suggestion above has it at `line 25`. At the point of `line 28` where you currently have the check, the function would have already returned due to what's currently at `line 25`.

Also, I thought about it, and the `string(data) != state` error should be something more explicit like `invalid oauth state`.

Reply:
Fixed this, oversight from my end

Reply:
Sorry for being so persistent about this (it's especially a pain because it's a small PR), but what do you think about changing the error string for the `string(data) != state` check to be `invalid oauth state`?

Reply:
No worries, Done